lib/saml/auth_fail_handler.rb

```diff
@@ -42,24 +43,28 @@ def generic_error_message
       context = {
         saml_response: {
           status_message: @saml_response.status_message,
-          errors: @saml_response.errors
+          errors: @saml_response.errors,
+          code: @code
         }
       }
       set_sentry_params('Other SAML Response Error(s)', :error, context)
     end
 
     def clicked_deny?
       return false unless only_one_error? && @saml_response.status_message == CLICKED_DENY_MSG
+      @code = '001'
       set_sentry_params(CLICKED_DENY_MSG, :warn)
     end
 
     def auth_too_late?
       return false unless only_one_error? && @saml_response.errors[0].include?(TOO_LATE_MSG)
+      @code = '002'
       set_sentry_params(TOO_LATE_MSG, :warn, @saml_response.errors[0])
     end
 
     def auth_too_early?
       return false unless only_one_error? && @saml_response.errors[0].include?(TOO_EARLY_MSG)
+      @code = '003'
```

Comment:
this works for now, but I really liked the hash you had proposed...

maybe something like:

ERRORS = {
'clicked-deny' => { pattern: 'Subject did not consent to attribute release', code: '001' }
...
}

then you can just String#match on the pattern?

Reply:
🤷‍♀️ This was my attempt at changing as little as possible, since you didn't want me to do to much refactoring.  I'd like to do it like this (WIP) soon https://github.com/department-of-veterans-affairs/vets-api/compare/fix_too_early_late?expand=1